apps/smart-contracts/core/test/unit/LongShortToken.test.ts

```diff
@@ -45,4 +48,25 @@ describe('=> LongShortToken', () => {
       expect(await longShort.balanceOf(deployer.address)).to.eq(1)
     })
   })
+
+  describe('# burnFrom', () => {
+    beforeEach(async () => {
+      await longShort.connect(deployer).mint(user.address, TEST_AMOUNT)
+    })
+
+    it('allows token contract owner to burn user tokens without approval', async () => {
+      await longShort.connect(user).approve(deployer.address, 0)
+      expect(await longShort.connect(deployer).burnFrom(user.address, TEST_AMOUNT))
+        .emit(longShort.address, 'Transfer')
+        .withArgs(user.address, ZERO_ADDRESS, TEST_AMOUNT)
+      expect(await longShort.balanceOf(user.address)).eq(0)
+      expect(await longShort.totalSupply()).eq(0)
+    })
```

Comment:
Remember, we should try to keep tests in an Arrange/Act/Assert format unless its a one liner like a revert case.

Also, we should aim to not perform fragile checks like .eq(0) here because the checks here assume that the beginning supply would just be TEST_AMOUNT.

```suggestion
    it('allows token contract owner to burn user tokens without approval', async () => {
      const userBalanceBefore = await longShort.balanceOf(user.address)
      const totalSupplyBefore = await longShort.totalSupply()
      await longShort.connect(user).approve(deployer.address, 0)
      
      const tx = await longShort.connect(deployer).burnFrom(user.address, TEST_AMOUNT)
      
      expect(tx)
        .emit(longShort.address, 'Transfer')
        .withArgs(user.address, ZERO_ADDRESS, TEST_AMOUNT)
      expect(await longShort.balanceOf(user.address)).eq(userBalanceBefore.sub(TEST_AMOUNT))
      expect(await longShort.totalSupply()).eq(totalSupplyBefore.sub(TEST_AMOUNT))
    })
```